src/apis/contracts_api.ts

```diff
@@ -301,21 +302,21 @@ export class ContractsAPI {
         const simpleInterestTermsContract = await this.loadSimpleInterestTermsContract();
         const collateralizedSimpleInterestTermsContract = await this.loadCollateralizedSimpleInterestTermsContract();
 
-        const supportedTermsContracts = [
-            simpleInterestTermsContract,
-            collateralizedSimpleInterestTermsContract,
-        ];
+        const addressToContractType = new Map([
+            [
+                collateralizedSimpleInterestTermsContract.address,
+                TERMS_CONTRACT_TYPES.COLLATERALIZED_SIMPLE_INTEREST_LOAN,
+            ],
+            [simpleInterestTermsContract.address, TERMS_CONTRACT_TYPES.SIMPLE_INTEREST_LOAN],
+        ]);
```

Comment:
I think this can all be represented much more cleanly as a vanilla JS object using [fancy ES6 features](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Object_initializer#Computed_property_names)

Reply:
Done.